internal/k8s/client.go

```diff
@@ -107,13 +106,31 @@ var _ Client = K8sClient{}
 
 type PortForwarder func(ctx context.Context, restConfig *rest.Config, core apiv1.CoreV1Interface, namespace string, podID PodID, localPort int, remotePort int) (closer func(), err error)
 
+func ProvideK8sClient(ctx context.Context, env Env) (K8sClient, error) {
+	if env == EnvNone {
+		// No k8s, so no need to get any further configs
+		return K8sClient{env: env}, nil
```

Comment:
should we have some sort of ExplodingK8sClient that emits proper errors? i'm concerned that this will just emit panics from null pointer dereferences

Reply:
i.e., please make sure that you test the behavior where the user doesn't have `kubectl` but tries to use `k8s_yaml`, it seems like this will explode in that case